### PR TITLE
Check project code.

### PR DESCRIPTION
**Pull request by danill1278**
Hi, @didevgen , please, could you take a look at our project.

Theme: SmartHouse (Smart Kettle + Audio Speaker).
About project: 

1. We need to create two classes (Smart Kettle, Speaker) using Prototype OOP.
2. Put in the base class (Machine) common characteristics. 
3. And after that create other objects (SmartHouse) which will manage our classes teamwork.

Now process on the second step.

Files:
SmartKettle.txt - template for SmartKettle class
machineTemplate.js - template for Machine class
speakerTemplate.js - template for Speaker class
main.js - SmartKettle class realization
machine.js - Machine class realization
speaker.js - Speaker class realization

We use default esLinter + prettier settings, because we have not agreed on settings with our team.

Please, review our code.
@danill1278 , @alinalaktionova

### CODE REVIEW

File: SmartKettle.txt [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		+addWater(number): void
+		+get currentFullness(): Number
+		
+		
+		
+		
+
+
+
+
```
Comment (didevgen) [inline SmartKettle.txt]: What is that?

File: speaker.js [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+      if (
+        !(
+          "name" in track &&
+          typeof track.name === "string" &&
+          track.name &&
+          "duration" in track &&
+          typeof track.duration === "number" &&
+          track.duration > 0
+        )
+      ) {
```
Comment (didevgen) [inline speaker.js]: image

File: speaker.js [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+  if (this._currentTrack > 0) {
+    this._currentTrack--;
+  } else {
+    // if track is first in track list play last track
+    this._currentTrack = this._trackList.length - 1;
+  }
+
+  // setup new timer
+  this._startPlaying();
+};
```
Comment (didevgen) [inline speaker.js]: DRY

File: speaker.js [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+  this._stopPlaying();
+
+  // if lost time - rewind time more than 0, start playing from new time position
+  if (this._currentTimerValue - rewindTime > 0) {
+    this._startPlaying(this._currentTimerValue - rewindTime);
+  } else {
+    // if lost time - rewind time less than 0, start playing previouse song
+    this.previousTrack();
+  }
+};
```
Comment (didevgen) [inline speaker.js]: DRY

**Comment by danill1278** [comment]
Hi  @didevgen , we have fixed issues, please check.

File: machine.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+function Machine(name) {
+  if (this._checkNameValidity(name)) {
+    this._name = name;
```
Comment (didevgen) [inline machine.js]: this._name = name || 'default' or 

function Machine(name='default')

File: main.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    throw new Error("Turn on device, please!");
+  }
+  return true;
+};
+
+Machine.prototype._checkNameValidity = function(name) {
+  if (typeof name !== "string") {
+    throw new Error("Name must be a string");
+  }
+  const regex = /^[A-z0-9\s]{5,10}/i;
```
Comment (didevgen) [inline main.js]: same as above

**Pull request merged**